warehouse/accounts/services.py

```diff
@@ -618,6 +623,78 @@ def get_password_timestamp(self, user_id):
         user = self.get_user(user_id)
         return user.password_date.timestamp() if user.password_date is not None else 0
 
+    def check_device_valid(self, user_id: str, device_id_secret_b64: str) -> bool:
+        """
+        Checks whether the given device id and secret are valid for the given user.
+        """
+        self._metrics.increment("warehouse.authentication.check_device_valid.start")
+
+        if not device_id_secret_b64:
```

Comment:
I'm assuming this is a None check? If so,

```suggestion
        if device_id_secret_b64 is None:
```